libraries/core_libs/consensus/src/vote_manager/vote_manager.cpp

```diff
@@ -471,6 +492,180 @@ uint64_t VoteManager::roundDeterminedFromVotes(size_t two_t_plus_one) {
+  }
+
+  if (vote->getType() != cert_vote_type) {
+    LOG(log_er_) << "Reward vote type is not cert vote type " << *vote;
+    return false;
+  }
+
+  if (vote->getStep() != 3) {
+    LOG(log_er_) << "Reward vote step is not 3 " << *vote;
+    return false;
+  }
+
+  const uint64_t dpos_period = pbft_chain_->getPbftChainSize() - 1;  // reward period - 1
+  const auto& voter_account_addr = vote->getVoterAddr();
+  uint64_t voter_dpos_votes_count;
+  try {
+    voter_dpos_votes_count = final_chain_->dpos_eligible_vote_count(dpos_period, voter_account_addr);
+  } catch (state_api::ErrFutureBlock& c) {
+    LOG(log_er_) << c.what() << ". Voter account " << voter_account_addr << " in DPOS period<" << dpos_period
+                 << "> is too far for ahead of DPOS. Have executed chain size " << final_chain_->last_block_number();
+    return false;
+  }
+
+  if (!voter_dpos_votes_count) {
+    LOG(log_er_) << "Invalid reward vote: voter was not eligible to vote " << *vote;
+    return false;
+  }
+
+  const auto dpos_total_votes_count = final_chain_->dpos_eligible_total_vote_count(dpos_period);
+  const size_t pbft_sortition_threshold = std::min<size_t>(pbft_committee_size_, dpos_total_votes_count);
+  try {
+    vote->validate(voter_dpos_votes_count, dpos_total_votes_count, pbft_sortition_threshold);
+  } catch (const std::logic_error& e) {
+    LOG(log_er_) << e.what();
+    LOG(log_er_) << "DPOS period " << dpos_period << ", voter " << voter_account_addr << " votes count "
+                 << voter_dpos_votes_count << ", total votes count " << dpos_total_votes_count
+                 << ", pbft sortition threshold " << pbft_sortition_threshold;
+
+    return false;
+  }
+
+  return true;
+}
+
+std::vector<std::shared_ptr<Vote>> VoteManager::updateRewardVotes(uint64_t reward_period) {
+  // TODO[issue1699]
+  auto period_data = db_->getPeriodDataRaw(reward_period);
+  if (period_data.size() == 0) {
+    // First period no reward votes
+    return {};
+  }
+
+  SyncBlock reward_period_sync_block(period_data);
+
+  std::unordered_set<vote_hash_t> reward_period_cert_votes_hashes;
+  for (const auto& v : reward_period_sync_block.cert_votes) {
+    reward_period_cert_votes_hashes.insert(v->getHash());
+  }
+
+  bool update = false;
+  {
+    const auto reward_period_pbft_block_hash = reward_period_sync_block.pbft_blk->getBlockHash();
+    std::unique_lock lock(reward_votes_mutex_);
+    if (reward_votes_.first != reward_period_pbft_block_hash) {
+      LOG(log_dg_) << "Reward period pbft block<" << reward_period_pbft_block_hash
+                   << "> does not match reward votes voted value " << reward_votes_.first;
+      return reward_period_sync_block.cert_votes;
+    }
+    for (auto& v : reward_votes_.second) {
+      if (!reward_period_cert_votes_hashes.contains(v.first) && verifyRewardVote(v.second)) {
+        LOG(log_nf_) << "Add new reward vote " << v.first;
+        update = true;
+        reward_period_cert_votes_hashes.insert(v.first);
```

Comment:
why you inserting it to that set? you do not use is afterwards anywhere

Reply:
I use it https://github.com/Taraxa-project/taraxa-node/pull/1670/files#diff-94ad8b2f7869e020e3e5b2226741b29fb3cb02d7f1c13e29012c17744eba1558R617

Reply:
Fixed, Thanks